Foreign fields pt 2 (#777)
o1-labs/docs2: docs/zkapps/o1js-reference/classes/Unconstrained.md

[o1js](../README.md) / [Exports](../modules.md) / Unconstrained

# Class: Unconstrained<T\>

> Container which holds an unconstrained value. This can be used to pass values

Reviewer: ```suggestion
Container which holds an unconstrained value. You can use this to pass values
```

Reviewer: I don't know if we want to make the API docs actionable, but if we did, this would be a valid suggestion

between the out-of-circuit blocks in provable code.

Invariants:

> - An `Unconstrained`'s value can only be accessed in auxiliary contexts.

Reviewer: ```suggestion
- An `Unconstrained`'s value can be accessed only in auxiliary contexts.
```
